api/lib/domain/usecases/authentication/authenticate-oidc-user.js

```diff
@@ -71,3 +78,15 @@ async function _updateAuthenticationMethodWithComplement({
     identityProvider: oidcAuthenticationService.identityProvider,
   });
 }
+
+async function _checkUserAccessScope({ audience, userId, adminMemberRepository }) {
+  if (audience === PIX_ADMIN.AUDIENCE) {
+    const adminMember = await adminMemberRepository.get({ userId });
+    if (!adminMember?.hasAccessToAdminScope) {
+      throw new ForbiddenAccess(
+        'User does not have the rights to access the application',
+        'PIX_ADMIN_ACCESS_NOT_ALLOWED',
+      );
+    }
+  }
+}
```

Comment:
J'ai fait un refacto et indiqué dedans des commentaires pour évoquer les points qui m'ont amenés à tenter une ré-écriture

```suggestion
// Proposition de renommage car cette fonction est très spécialisée
async function _assertUserWithPixAdminAccess({ audience, userId, adminMemberRepository }) {
  if (audience !== PIX_ADMIN.AUDIENCE) {
    return; // Il manquerait pas un throw pour traiter ce cas ?
  }

  // Le elvis operator '?' me semble superflu, le repository est pas censé renvoyer null ou undefined puisque
  // ce n'est pas un find
  const hasAccessToAdminScope = (await adminMemberRepository.get({ userId }).hasAccessToAdminScope) ?? false;
  if (!hasAccessToAdminScope) {
    // En faire une DomainError puisque qu'on est dans le domain ?
    throw new ForbiddenAccess(
      'User does not have the rights to access the application',
      'PIX_ADMIN_ACCESS_NOT_ALLOWED',
    );
  }
}
```

Reply:
```
if (audience !== PIX_ADMIN.AUDIENCE) {
    return; // Il manquerait pas un throw pour traiter ce cas ?
  }
```

C'est volontaire de notre part de ne pas ajouter cette condition car ce usecase est utilisé à l'origine par Pix App.
Si on retrouve l'utilisateur en BDD (table "users") alors on l'authentifie.
Mais dans le cas de Pix Admin, en plus d'identifier l'utilisateur, il faut qu'il ait un accès avec un rôle (table "pix-admin-roles") pour accéder à l'application.

Donc si on est dans le cas Pix App, on cherche à ignorer cette vérification et à faire comme d'habitude. Mais si on est sur Pix Admin, on veut faire ce check supplémentaire avant authentification

Reply:
```
// En faire une DomainError puisque qu'on est dans le domain ?
    throw new ForbiddenAccess(
      'User does not have the rights to access the application',
      'PIX_ADMIN_ACCESS_NOT_ALLOWED',
    );
```

Sauf s'il me manque une info mais c'est une domain error déjà

Reply:
Donc de ta proposition ce que je pense garder 

```
async function _assertUserWithPixAdminAccess({ audience, userId, adminMemberRepository }) {
  if (audience === PIX_ADMIN.AUDIENCE) {
    const hasAccessToAdminScope = (await adminMemberRepository.get({ userId }).hasAccessToAdminScope) ?? false;
    if (!hasAccessToAdminScope) {
      throw new ForbiddenAccess(
        'User does not have the rights to access the application',
        'PIX_ADMIN_ACCESS_NOT_ALLOWED',
      );
    }
  }
}
```

cette proposition te convient @clemlatz ?

Reply:
@Steph0 j'aime bien l'idée du "return early", par contre le nom de la méthode `get` t'a induit en erreur mais elle peut bien retourner `undefined`, donc pour le reste nous allons garder le code actuel.